pkg/controller/direct/monitoring/refs.go

```diff
@@ -101,5 +186,21 @@ func (r *refNormalizer) VisitField(path string, v any) error {
 			}
 		}
 	}
+	if alertChart, ok := v.(*krm.AlertChart); ok {
+		if ref, err := normalizeMonitoringAlertPolicyRef(r.ctx, r.kube, r.src, r.project, alertChart.AlertPolicyRef); err != nil {
+			return err
+		} else {
+			alertChart.AlertPolicyRef = ref
+		}
+	}
+
+	if projectRef, ok := v.(*refs.ProjectRef); ok {
```

Comment:
This is sneaky ... these are still fields!  So we could actually have `refs.ProjectRef` implement a Normalize interface, and then we have references just work automatically with no extra controller code.